src/colorizer/ColorizeCanvas.ts

```diff
@@ -175,18 +175,17 @@ export default class ColorizeCanvas {
   }
 
   setFeature(name: string): void {
-    const featureData = this.dataset?.getFeatureData(name);
-    if (!featureData) {
+    if(!this.dataset?.hasFeature(name)) {
       return;
     }
+    const featureData = this.dataset?.getFeatureData(name)!;
```

Comment:
Style question here! I changed this so it calls `hasFeature` and added the `!` symbol to assert that the feature data isn't null, but I wanted feedback on if this is a good change or not.

I like that it's slightly clearer to read (it's checking that the feature exists, then getting its data), but it might be possible for the dataset to have bad feature data? Is the original check better here?

Reply:
Yeah, I think this is good!

To potentially overexplain my philosophy here: the idea of explicitly typing `getFeatureData` to maybe return `null` is to have the type system force you to handle whatever errors are possible before using the value. In this context, `!` means "I checked, no errors are possible." Like Dan said I don't think "bad feature data" is a thing yet, or will ever be since we make all our own data, so checking if a feature with that key exists is all you need to do to make that assertion.

Reply:
In fact, you can turn the `?.` after `dataset` into a regular old `.` here, since typescript recognizes that we wouldn't pass the if statement above if `dataset` were undefined, and narrows its type for the rest of the function

Reply:
Okay, great, thank you! Marking this as resolved 😊